Mathlib/Init/Logic.lean

```diff
@@ -539,15 +539,64 @@ theorem right_comm : Commutative f → Associative f → RightCommutative f :=
 
 end Binary
 
+namespace Acc
+
+universe v u
+variable {α : Sort u} {r : α → α → Prop}
+
+@[elab_as_elim]
+unsafe def recU {motive : (a : α) → Acc r a → Sort v}
```

Comment:
@digama0 Could you please take a look at this

Reply:
There is no need for this to have an unsafe implementation. It can be defined using `termination_by` and still be computable.

But also: this should be a PR to core. If that stalls we can consider this workaround.

Reply:
I don't think is going to make it into core any time soon, and almost certainly not in the csimp version.

Instead of mathlib4, we could put it in std4 though.

Reply:
The unsafe `implemented_by` version seems like a very simple drop in thing. I would not recommend the csimp version anyway, but adding this to the Acc.rec that lives in core means that it will also work for any definitions depending on it as well, rather than only the ones downstream of whatever redefinition is happening in std4/mathlib4.

Reply:
I mean sure, I agree that it would be better if `Acc.rec` (and other `.rec`s) compiled out of the box.  But realistically I don't see that happening soon.  In the meantime we should put it here.

In the end, this is just an additional feature.  And std4 is the place where additional features go that core doesn't need.

Reply:
I made `Acc.rec` computable without `implemented_by`.
I will soon create a PR in std to move these workarounds.

Reply:
@digama0 Please give me permission to publish branch in std.

Reply:
@Komyyy std uses the traditional github fork-and-PR model, so you shouldn't need write access.

Reply:
I created the PR: leanprover/std4#123